scripts/eval-with-monitoring.sh

```
@@ -0,0 +1,110 @@
+echo "Monitor log: $MONITOR_LOG"
+echo ""
+
+# Pre-eval snapshot
+echo "--- PRE-EVAL SNAPSHOT ---"
+snapshot_processes "PRE-EVAL"
+echo ""
+
+# Start background monitor
+background_monitor &
+MONITOR_PID=$!
+trap "kill $MONITOR_PID 2>/dev/null; wait $MONITOR_PID 2>/dev/null" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)